dot/parachain/network-bridge/sender_test.go

```diff
@@ -0,0 +1,202 @@
+		require.NoError(t, err)
+
+		result := <-request.Result
+		require.Error(t, result.Error)
+		require.Nil(t, result.Response)
+		requireClosed(t, request.Result)
+	})
+
+	t.Run("cancel_request", func(t *testing.T) {
+		t.Parallel()
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		request := makeOutgoingRequest(t)
+		nbs := setUpNetworkBridgeSender(t, ctrl, request, nil, nil, nil)
+
+		sendRequests := networkbridgemessages.SendRequests{
+			Requests:       []*networkbridgemessages.OutgoingRequest{request},
+			IfDisconnected: networkbridgemessages.TryConnect,
+		}
+
+		request.Cancel()
+
+		err := nbs.processMessage(sendRequests)
+		require.NoError(t, err)
+
+		result := <-request.Result
+		require.Nil(t, result.Response)
+		require.NoError(t, result.Error)
+		requireClosed(t, request.Result)
+	})
+}
+
+// We arbitrarily use a ChunkFetchingRequest since it does not matter for testing SendRequests handling.
+func makeOutgoingRequest(t *testing.T) *networkbridgemessages.OutgoingRequest {
+	t.Helper()
+
+	return networkbridgemessages.NewOutgoingRequest(
+		"recipient",
+		networkbridgemessages.ChunkFetchingRequest{
+			CandidateHash: parachaintypes.CandidateHash{Value: common.Hash{1}},
+			Index:         42,
+		})
+}
+
+// Expect calls to Network.GetRequestResponseProtocol() and RequestMaker.Do() when only one of response, rawResponse or
+// reqErr is non-nil.
+// Expect no calls Network.GetRequestResponseProtocol() RequestMaker.Do() when all three are nil.
+func setUpNetworkBridgeSender(
+	t *testing.T,
+	ctrl *gomock.Controller,
+	request *networkbridgemessages.OutgoingRequest,
+	response network.ResponseMessage,
+	rawResponse []byte,
+	reqErr error,
+) *NetworkBridgeSender {
+	t.Helper()
+
+	expectCancellation := response == nil && rawResponse == nil && reqErr == nil
+	reqMaker := NewMockRequestMaker(ctrl)
+
+	if expectCancellation {
+		reqMaker.EXPECT().
+			Do(gomock.Any(), gomock.Any(), gomock.Any()).
+			Times(0)
+	} else {
+		reqMaker.EXPECT().
+			Do(request.Recipient, request.Payload, gomock.AssignableToTypeOf(request.Payload.Response())).
+			DoAndReturn(func(to peer.ID, req network.Message, res network.ResponseMessage) error {
+				if reqErr != nil {
+					return reqErr
+				}
+
+				if response != nil {
+					var err error
+					rawResponse, err = response.Encode()
+					require.NoError(t, err)
+				}
+
+				if err := res.Decode(rawResponse); err != nil {
+					return err
+				}
+				return nil
+			})
+	}
+
+	netService := NewMockNetwork(ctrl)
+
+	if expectCancellation {
+		netService.EXPECT().
+			GetRequestResponseProtocol(gomock.Any(), gomock.Any(), gomock.Any()).
+			Times(0)
+	} else {
+		netService.EXPECT().
+			GetRequestResponseProtocol(request.Payload.Protocol().String(), gomock.Any(), gomock.Any()).
+			Return(reqMaker)
+	}
+
+	return RegisterSender(nil, netService)
+}
+
+func requireClosed(t *testing.T, ch chan networkbridgemessages.ReqRespResult) {
+	select {
+	case <-ch:
```

Comment:
just to add an extra layer, when the channel is closed `ok` is false. Actually, you don't need the select when you retrieve a tuple from the channel (even if the channel is not closed)
```suggestion
 _, ok := <-ch
require.False(t, ok)
```

Reply:
If the channel is empty but not closed, the receive *will* block though. The `ok` merely indicates whether the received value is the zero value for the type of the channel, i.e. whether the channel was empty.